Week2/js/HomeTask2.js

```diff
@@ -0,0 +1,204 @@
+    logoImage.setAttribute("src", logoPaths[i]);
+    logoImage.setAttribute("class", "logo-image");
+    logo.appendChild(logoImage);
+    let content = document.createElement("div");
+    content.setAttribute("class", "content");
+    let title = document.createElement("h3");
+    title.innerText = titleNames[i];
+    let description = document.createElement("p");
+    description.innerText =
+      "We have helped over 120 Fortune 1000 companies in the following" +
+      "industries solve their most complex technology challenges. Now see " +
+      "what we can do for you. We have helped over 120 Fortune 1000 companies" +
+      "in the following industries solve their most complex technology" +
+      "challenges. Now see what we can do for you. We have helped over 120" +
+      "Fortune 1000 companies in the following industries solve their most" +
+      "complex technology challenges. Now see what we can do for you";
+    let clickToReadMore = document.createElement("button");
+    clickToReadMore.setAttribute("value", "Continue Reading");
+    clickToReadMore.setAttribute("class", "continue-reading");
+    clickToReadMore.setAttribute("id", i + 1);
+    clickToReadMore.addEventListener("click", showPoPUp);
+    clickToReadMore.innerText = "Continue Reading";
+    content.appendChild(title);
+    content.appendChild(description);
+    content.appendChild(clickToReadMore);
+    mainContainer.appendChild(logo);
+    mainContainer.appendChild(content);
+    divisions[i] = mainContainer;
+  }
+  return divisions;
+};
+var channels = generateItems();
+
+var createCategories = function() {
+  let selectCategory = document.createElement("select");
+  let channelNames = [
+    "Canada",
+    "Mexico",
+    "UnitedStates",
+    "Austria",
+    "Bulgaria",
+    "Czech Republic",
+    "Germany",
+    "Hungary",
+    "Ireland",
+    "Netherlands"
+  ];
+  selectCategory.setAttribute("class", "dropdown");
+  selectCategory.setAttribute("id", "dropdown");
+  selectCategory.addEventListener("change", displaySelectedItem);
+  var category = document.createElement("option");
+  category.setAttribute("value", "ALL");
+  category.innerText = "ALL";
+  selectCategory.appendChild(category);
+  for (var i = 0; i < 10; i++) {
+    var category = document.createElement("option");
+    category.setAttribute("value", i + 1);
+    category.innerText = channelNames[i];
+    selectCategory.appendChild(category);
+  }
+  return selectCategory;
+};
+
+var displayAll = function(dataDiv) {
+  channels.forEach(element => {
+    dataDiv.appendChild(element);
+  });
+  mainDivision.appendChild(dataDiv);
+};
+
+var removeAll = function() {
+  let currentDataDiv = document.getElementById("data-div");
+  let items = currentDataDiv.childNodes;
+  var child = currentDataDiv.lastElementChild;
+  while (child) {
+    currentDataDiv.removeChild(child);
+    child = currentDataDiv.lastElementChild;
+  }
+  while (currentDataDiv.hasChildNodes()) {

```

Comment:
Instead of removing elements from DOM, why can't we show/hide?

Reply:
ya we can do it in that way also , but my friend said that removing element from DOM is standard way of doing it . so i did it in that way. First i did show/hide

Reply:
If the functionality is to show all and then to hide few then show/hide is good. If you show few first and then all then adding only the blocks is a good practice.(like ngIf in angular which you will learn in future)